server/rpc.go

```diff
@@ -305,3 +305,46 @@ func pmTicketParams(params *net.TicketParams) *pm.TicketParams {
 		},
 	}
 }
+
+func coreSegMetadata(segData *net.SegData) (*core.SegTranscodingMetadata, error) {
+	if segData == nil {
+		glog.Error("Empty seg data")
+		return nil, errors.New("empty seg data")
+	}
+	var err error
+	profiles := []ffmpeg.VideoProfile{}
+	if len(segData.FullProfiles2) > 0 {
+		profiles, err = makeFfmpegVideoProfiles(segData.FullProfiles2)
+	} else if len(segData.FullProfiles) > 0 {
+		profiles, err = makeFfmpegVideoProfiles(segData.FullProfiles)
+	} else if len(segData.Profiles) > 0 {
+		profiles, err = common.BytesToVideoProfile(segData.Profiles)
+	}
+	if err != nil {
+		glog.Error("Unable to deserialize profiles ", err)
+		return nil, err
+	}
+
+	var os *net.OSInfo
+	if len(segData.Storage) > 0 {
+		os = segData.Storage[0]
+	}
+
+	dur := time.Duration(int64(segData.Duration) * int64(time.Millisecond))
```

Comment:
```suggestion
	dur := time.Duration(segData.Duration) * time.Millisecond
```

Reply:
Fixed in b7a6479c1ec41cf2a373d629eff8d1a687d40ec7